smolyay/basis.py

```diff
@@ -86,18 +102,8 @@ class ChebyshevFirstKind(BasisFunction):
     """
 
     def __init__(self, n):
-        super().__init__()
+        super().__init__([-1, 1])
         self._n = n
```

Comment:
Does `n` need to be readonly now? It seems like the class will work fine if you modify n. If so, you might want to just do `self.n = n` and remove the property below.

Reply:
After you put the BasisFunction in a set, there's no reason to change n after initialization. So making it readonly protects it from being changed. If that's not very pythonic, then we can change it.

Reply:
Someone should have enough common sense not to change `n` in that case, so I vote to make it mutable.